h2o-core/src/main/java/water/H2O.java

```diff
@@ -1915,6 +1935,13 @@ public static void main( String[] args ) {
     // join an existing Cloud.
     new HeartBeatThread().start();
 
+    // register disconnect hooks for the disconnect watchdog
+    registerNodeDisconnectedHook(new NodeDisconnectHook.ClientDisconnectedHook());
```

Comment:
Sure. Right now I'm looking at the code how `AbstractExtension` is used. If I'm correct than I should do call `H2O.addExtension(extension)` in the `H2O`'s main method